src/index.js

```diff
@@ -114,12 +116,49 @@ export default class CSSSplitWebpackPlugin {
           return new SourceMapSource(
             css,
             name(i),
-            map.toString()
+            map && map.toString()
           );
         }),
       });
     });
   }
+  
+  chunksMapping(chunks, assets, publicPath) {
```

Comment:
You can add `done` to the list of args to and DRY up some duplicate code you have in the below section:

```
Promise.all(promises).then(() => {
  done();
}, done);
```

Reply:
You may also wish to pass `compilation` instead of `publicPath` as you have some duplicate `publicPath` code below as well.